flexget/plugins/filter/exists_movie.py

```diff
@@ -74,54 +94,76 @@ def on_task_filter(self, task, config):
             #logging.getLogger('imdb_lookup').setLevel(logging.WARNING)
 
             # scan through
-
-            # TODO: add also video files?
-            for item in folder.walkdirs(errors='ignore'):
-                if item.name.lower() in self.skip:
-                    continue
-                count_dirs += 1
-
-                movie = get_plugin_by_name('parsing').instance.parse_movie(item)
-
-                try:
-                    imdb_id = imdb_lookup.imdb_id_lookup(movie_title=movie.name,
-                                                         raw_title=item.name,
-                                                         session=task.session)
-                    if imdb_id in path_ids:
-                        log.trace('duplicate %s' % item)
+            for root, dirs, files in os.walk(folder):
+                for item in eval(config['type']):
```

Comment:
I switched to `os.walk()` because it iterates over both files and dirs in the same function instead of using `path.walkfiles()` and `path.walkdirs()`. Combined with the 'eval(config['type'])` to choose what variable `dirs/files` to iterate over, it seemed like a good way to avoid repetition.

However, when adding the patterns to skip (which are different for files and dirs) it wasn't so nice anymore. So I was planning on rewriting this anyway.